### ISSUE TITLE
Feature request: Nightmode

### ISSUE THREAD

**Issue by trivago-dcalvo** [prymitive/karma#371]
First of all, thank you for developing this dashboard! It is very handy for us at trivago.

Would it be possible to have a "Nightmode" or "Dark mode" setting in the dashboard?

Some of us work in a low luminosity environment and the karma dashboard is bright. It strains the eye after a while.

If the little alarm windows could have a dark gray background for example, and the window could have an even darker shade of gray in this option, that would be great! Grafana does this very well.

Do you think this would be possible? It would certainly make some of your users very happy!

**Comment by prymitive** [comment]
Hi, this is a similar request to #189, the only problem for me is that UI customizations like this makes it a harder to make changes and test them (visual UI testing isn't easy if the UI look is conditional), so I was hoping that allowing to use custom css would allow to fine tune it.

The only problem with custom css approach is that on occasion it might break, since karma uses [bootstrap](https://getbootstrap.com) it follows classes that bootstrap exposes, if those change there's a possibility that custom css might stop working correctly. But this is rare so it basically means that some karma upgrades might require adjusting custom css (like add an extra class). 

An example:

1. Create custom.css file, here's the bare minimum, tweak the color value as you like:

```css
.card,
.list-group-item,
.dropdown-menu,
.modal-content {
    background-color: #2E2727!important;
}
```

2. Tell karma to serve it to users:

```
$ karma --custom.css /path/to/custom.css
```

Result:
![screencapture-localhost-8080-2019-01-17-12_12_47](https://user-images.githubusercontent.com/114288/51318091-3d9ec800-1a51-11e9-8c15-47b4ff325e39.png)

Would that work for you?

**Comment by prymitive** [comment]
Another problem is that while karma itself uses bootstrap directly, there are some widgets (like [react-select](https://react-select.com/) used on the silence form) that implement a different way of styling components, which doesn't allow to easily use bootstrap classes there and as a dirty had I resorted to hardcoding css on those. So there are definitely challenges to implementing a true night shift / dark mode.
But I think I could just add a few example custom css files to this repo and the docker image, so instead of writing your own you could just  pass it to karma itself. This way someone who wants a dark mode wouldn't need to write anything (unless they want to customise it) while keeping that mode as separate feature (no part of the main code base).
Would you be happy with that?

**Comment by trivago-dcalvo** [comment]
@prymitive I would be very happy with the custom css solution you offered. I believe anyone wanting a nightmode / different application appearance would also be happy with it.

We're currently running your application as a container, so passing the nightmode css as that as a container argument would be the icing on the cake for us. Something like:

`docker run --detach --restart always -p 8080:8080 -e CUSTOM_CSS=nightmode.css -e ALERTMANAGER_URI=http://192.168.1.50:9093 lmierzwa/karma:latest`

Thank you for your detailed answers on the subject and willingness to help!

**Comment by prymitive** [comment]
👍 I'll add a more complete theme before the next release

**Comment by prymitive** [comment]
How about this one?

```
body,
.bg-primary-transparent {
  background-color: #2E2727!important;
}

.card,
.list-group-item,
.dropdown-menu,
.modal-content {
    background-color: #455a64!important;
}


.input-group-text,
.components-filterinput {
  background-color: #455a64!important;
}

.components-grid-annotation {
  background-color: #95a5a6!important;
}

.badge:not(.components-label-bright),
.form-group > label,
.nav-link {
  color: #ccc!important;
}

.nav > .nav-link {
  color: #ccc!important;
}

.nav > .nav-link.active,
.nav-item > .nav-link.active {
  color: #fff!important;
  background-color: #455a64!important;
}

.badge.badge-light {
  color: #666;
  background-color: #444;
}
```

**Comment by prymitive** [comment]
Added it via #398, you can try it by passing `CUSTOM_CSS=/themes/dark.css` to your docker instance (if you pull `karma:latest` in ~30 minutes). Let me know if that works

**Comment by trivago-dcalvo** [comment]
It's on! 

![](https://i.imgur.com/3Cuggdb.jpg)

Thank you very much Łukasz. If you ever visit Mallorca, the beers are on me.

**Comment by prymitive** [comment]
I've added experimental dark mode in [v0.49](https://github.com/prymitive/karma/releases/tag/v0.49) so the old css is gone now.
The code is ugly and I'm not exactly very happy with it, but a "proper" implementation requires me to move all style code from (s)css to some css-in-js implementation (like styled components), which would be a big task. Will get there some day, but for now some ugly css hacking will do.

**Comment by prymitive** [comment]
storybook and https://percy.io/ made this much easier to implement and maintain, so I've merged a  "proper" dark mode in #1181

**Comment by prymitive** [comment]
Some new visual testing tooling made it easier to maintan 2 versions of the color scheme, but loading 2 different css files comes with a few problems so I wanted to ensure it doesn't degrade on "not my browser".
@trivago-dcalvo are you able to test the latest docker image and check if the new dark mode works without any quirks?

**Comment by prymitive** [comment]
This will now follow browser preference by default, so if you run macOs or iOS and enable dark mode for the UI karma will switch to dark theme.

**Comment by prymitive** [comment]
Released [v0.52](https://github.com/prymitive/karma/releases/tag/v0.52) with the new dark mode.

**Comment by prymitive** [comment]
Seems to be working well, so closing
